### PR TITLE
Wasm staging checklist

### PR DESCRIPTION
**Pull request by gahaas**
I want to add a checklist of requirements for staging and shipping a WebAssembly feature.

### CODE REVIEW

File: src/docs/wasm-shipping-checklist.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: 'Checklist for Staging and Shipping of WebAssembly features'
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
title: 'Checklist for staging and shipping of WebAssembly features'
```

File: src/docs/wasm-shipping-checklist.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+description: 'This document provides checklists of engineering requirements on when to stage and ship a WebAssembly feature in V8.'
+---
+This document should provide checklists of engineering requirements on when to
+stage and when to ship a WebAssembly feature in V8. These checklists are meant
+as a guideline and may not be applicable to all features. The actual launch
+process is described in the [V8 Launch process](https://v8.dev/docs/feature-launch-process).
+
+## Staging
+
+### When to stage a WebAssembly Feature
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
### When to stage a WebAssembly feature
```

File: src/docs/wasm-shipping-checklist.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+The [staging](https://docs.google.com/document/d/1ZgyNx7iLtRByBtbYi1GssWGefXXciLeADZBR_FxG-hE) of a WebAssembly feature defines the end of its implementation phase. The implementation phase is finished when the following checklist is done:
+* The implementation in V8 is complete. This includes:
+  * Implementation in TurboFan (if applicable)
+  * Implementation in Liftoff (if applicable)
+  * Implementation in the interpreter (if applicable)
+* Tests in V8 are available
+* Spec tests are rolled into V8 by running [./tools/wasm/update-wasm-spec-tests.sh](https://cs.chromium.org/chromium/src/v8/tools/wasm/update-wasm-spec-tests.sh)
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
* Spec tests are rolled into V8 by running [`tools/wasm/update-wasm-spec-tests.sh`](https://cs.chromium.org/chromium/src/v8/tools/wasm/update-wasm-spec-tests.sh)
```

File: src/docs/wasm-shipping-checklist.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+* Pass all existing proposal spec tests. Missing spec tests are unfortunate but should not block staging.
+
+Note that the stage of the feature proposal in the standardization process does not matter for staging the feature in V8. The proposal should, however, be mostly stable.
+
+### How to stage a WebAssembly Feature
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
### How to stage a WebAssembly feature
```

File: src/docs/wasm-shipping-checklist.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+* In [src/wasm/wasm-feature-flags.h](https://cs.chromium.org/chromium/src/v8/src/wasm/wasm-feature-flags.h), move the feature flag from the `FOREACH_WASM_EXPERIMENTAL_FEATURE_FLAG` macro list to the `FOREACH_WASM_STAGING_FEATURE_FLAG` macro list.
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
* In [`src/wasm/wasm-feature-flags.h`](https://cs.chromium.org/chromium/src/v8/src/wasm/wasm-feature-flags.h), move the feature flag from the `FOREACH_WASM_EXPERIMENTAL_FEATURE_FLAG` macro list to the `FOREACH_WASM_STAGING_FEATURE_FLAG` macro list.
```

Comment (mathiasbynens) [reply src/docs/wasm-shipping-checklist.md]: similar for the list items below

File: src/docs/wasm-shipping-checklist.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+### When is a WebAssembly feature ready to be shipped
+
+* The [V8 Launch process](https://v8.dev/docs/feature-launch-process) is satisfied.
+* The feature has been staged for several weeks to get fuzzer coverage.
+* The feature proposal is [stage 4](https://github.com/WebAssembly/proposals).
+* All [spec tests](https://github.com/WebAssembly/spec/tree/master/test) pass.
+* The implementation is covered by a fuzzer (if applicable).
+
+### How to ship a WebAssembly Feature
+* In [src/wasm/wasm-feature-flags.h](https://cs.chromium.org/chromium/src/v8/src/wasm/wasm-feature-flags.h), move the feature flag from the `FOREACH_WASM_STAGING_FEATURE_FLAG` macro list to the `FOREACH_WASM_SHIPPED_FEATURE_FLAG` macro list.
```
Comment (mathiasbynens) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
* In [`src/wasm/wasm-feature-flags.h`](https://cs.chromium.org/chromium/src/v8/src/wasm/wasm-feature-flags.h), move the feature flag from the `FOREACH_WASM_STAGING_FEATURE_FLAG` macro list to the `FOREACH_WASM_SHIPPED_FEATURE_FLAG` macro list.
```

**Comment by gahaas** [comment]
I addressed all comments, and ran `npm run lint`

File: src/docs/wasm-shipping-checklist.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+title: 'Checklist for staging and shipping of WebAssembly features'
+description: 'This document provides checklists of engineering requirements on when to stage and ship a WebAssembly feature in V8.'
+---
+This document should provide checklists of engineering requirements on when to
+stage and when to ship a WebAssembly feature in V8. These checklists are meant
```
Comment (ecmziegler) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
This document provides checklists of engineering requirements for staging and shipping WebAssembly features in V8. These checklists are meant
```

File: src/docs/wasm-shipping-checklist.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+The [staging](https://docs.google.com/document/d/1ZgyNx7iLtRByBtbYi1GssWGefXXciLeADZBR_FxG-hE) of a WebAssembly feature defines the end of its implementation phase. The implementation phase is finished when the following checklist is done:
+
+- The implementation in V8 is complete. This includes:
+    - Implementation in TurboFan (if applicable)
+    - Implementation in Liftoff (if applicable)
+    - Implementation in the interpreter (if applicable)
+- Tests in V8 are available
+- Spec tests are rolled into V8 by running [`tools/wasm/update-wasm-spec-tests.sh`](https://cs.chromium.org/chromium/src/v8/tools/wasm/update-wasm-spec-tests.sh)
+- Pass all existing proposal spec tests. Missing spec tests are unfortunate but should not block staging.
```
Comment (ecmziegler) [inline src/docs/wasm-shipping-checklist.md]:
```suggestion
- All existing proposal spec tests pass. Missing spec tests are unfortunate but should not block staging.
```

File: src/docs/wasm-shipping-checklist.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+# Shipping
+
+## When is a WebAssembly feature ready to be shipped
+
+- The [V8 Launch process](https://v8.dev/docs/feature-launch-process) is satisfied.
+- The feature has been staged for several weeks to get fuzzer coverage.
+- The feature proposal is [stage 4](https://github.com/WebAssembly/proposals).
+- All [spec tests](https://github.com/WebAssembly/spec/tree/master/test) pass.
+- The implementation is covered by a fuzzer (if applicable).
```
Comment (ecmziegler) [inline src/docs/wasm-shipping-checklist.md]: Move this item up before the one about fuzzer coverage?

Comment (gahaas) [reply src/docs/wasm-shipping-checklist.md]: Done.

File: src/docs/wasm-shipping-checklist.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+---
+title: 'Checklist for staging and shipping of WebAssembly features'
+description: 'This document provides checklists of engineering requirements on when to stage and ship a WebAssembly feature in V8.'
+---
+This document provides checklists of engineering requirements for staging and shipping WebAssembly features in V8. These checklists are meant
+as a guideline and may not be applicable to all features. The actual launch
```
Comment (RReverser) [inline src/docs/wasm-shipping-checklist.md]: Nit: these two line breaks look odd, we generally don't break lines on limits.

Comment (mathiasbynens) [reply src/docs/wasm-shipping-checklist.md]: +1. Line breaks are turned into hard `<br>`s, which would look bad on various viewport widths. On v8.dev, only use line breaks when you need one semantically.

Comment (gahaas) [reply src/docs/wasm-shipping-checklist.md]: Done. Could the linter check for that?

File: src/docs/wasm-shipping-checklist.md (hunk 5 above)
Comment (gahaas) [reply src/docs/wasm-shipping-checklist.md]: Done

File: src/docs/wasm-shipping-checklist.md (hunk 10 above)
Comment (mathiasbynens) [reply src/docs/wasm-shipping-checklist.md]: Hmm, I'm not sure. I don't think the linter has any way of knowing (for sure) whether the linebreak you added was intentional or not.

**Comment by RReverser** [comment]
You might also want to add a link to this page from `docs/index.md`. Otherwise it will be accessible only via direct link, which is probably not what you want?

**Comment by gahaas** [comment]
I added the link to index.md now

**Pull request merged**